src/test/java/no/unit/nva/handlers/AddRoleHandlerTest.java

```diff
@@ -99,6 +111,42 @@ public void statusCodeReturnsOkWhenRequestIsSuccessful() {
         assertThat(successCode, is(equalTo(HttpStatus.SC_OK)));
     }
 
+    @Test
+    public void addRoleHandlerThrowsUnexpectedExceptionWhenDatabaseServiceCannotFetchSavedRole()
+        throws InvalidRoleInternalException {
+
+        Class<InvalidRoleInternalException> expectedExceptionClass = InvalidRoleInternalException.class;
+        RoleDto inputRole = RoleDto.newBuilder().withName(SOME_ROLE_NAME).build();
```

Comment:
The creation of the tests could probably be refactored across this file to named methods that express the intent of the set up